src/GafferScene/IECoreScenePreview/PrimitiveAlgo.cpp

```diff
@@ -310,6 +310,76 @@ bool interpolationMatches(
 	}
 }
 
+PrimitiveVariable::Interpolation mergeInterpolations(
+	IECoreScene::TypeId primType, PrimitiveVariable::Interpolation a, PrimitiveVariable::Interpolation b,
+	const IECore::InternedString &msgName
+)
+{
+	PrimitiveVariable::Interpolation result;
+	if( a == PrimitiveVariable::Invalid || b == PrimitiveVariable::Invalid )
+	{
+		// Invalid inputs are always invalid.
+		result = PrimitiveVariable::Invalid;
+	}
+	else if( primType == IECoreScene::PointsPrimitiveTypeId )
+	{
+		// On points, everything is output as Vertex
+		result = PrimitiveVariable::Vertex;
+	}
+	else if( interpolationMatches( primType, a, b ) )
+	{
+		// If they're the same, then we don't need to promote.
+		result = a;
+	}
+	else if( a == PrimitiveVariable::Constant || ( primType != IECoreScene::MeshPrimitiveTypeId && a == PrimitiveVariable::Uniform ) )
```

Comment:
Oh, this is fine for correctness, but confusing: we later promote constant to uniform on Curves, since Constant wouldn't allow for different values from different source prims. So treating Constant as the same as Uniform is fine here. But sounds like you prefer the second version anyway.